### PR TITLE
Test optimizations

### LINKED ISSUE #309: Explore uses in tests
### Describe your feature request

Try to see if you can use uses to bind this to just this test case.

https://pestphp.com/docs/underlying-test-case#uses

### Describe the solution you'd like

for better optimisations for tests

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in this repo.

Yes

### Please confirm that you have searched in our documentation and couldn't find the answer.

Yes

### PR DESCRIPTION
**Pull request by dingo-d**
# Description

This PR makes several updates to the tests:

- Fixes the hardcoded directory paths in favor of the new `Components::getProjectPaths` helper
- Adds https://github.com/Yoast/wp-test-utils package that has the brainMonkey lib added as a dependency
- Clean up `composer.json` (reduce direct dependencies to reduce potential conflicts between the libs), group the test scripts
- Update workflow actions to the latest versions
- Added base test class that has the setup and teardown methods added to it, so that we don't need to call them in every test `beforeEach` and `afterEach` methods
- Minor fix in the components argument name (could be a BC break if we supported PHP 8, but we don't so, we're good)

This is just an optimization/cleanup PR. Can be released in the minor version.

Closes #309 

You can review this by commits. Some commits have lots of changes that are mostly removal of things, or move of the files (those can be skimmed over). Others are smaller and easier to review.

### CODE REVIEW

File: tests/Unit/Helpers/ErrorLoggerTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+use EightshiftLibs\Helpers\ErrorLoggerTrait;
+
+use function Tests\mock;
+
+class MockErrorLogger {
+	use ErrorLoggerTrait;
+}
+
+beforeEach(function () {
```
Comment (iruzevic) [inline tests/Unit/Helpers/ErrorLoggerTraitTest.php:13]: please unset the $this->mockLoggere after everything.

File: tests/Unit/Main/AutowiringTest.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	ServiceWithInterfaceDepMoreThanOneClassFound,
+	ServiceWithInterfaceDepWrongName,
+	ServiceWithMultipleDeps,
+	ServiceWithPrimitiveDep,
+	ServiceWithPrimitiveDepHasDefault
+};
+
+beforeEach(function() {
+
+	$this->main = new MainExample([
```
Comment (iruzevic) [inline tests/Unit/Main/AutowiringTest.php:35]: please unset this methods after the tests

**Review by iruzevic — CHANGES_REQUESTED** [review]
this looks awesome, just a few small things

**Comment by dingo-d** [comment]
@iruzevic I think I got all of them now.

**Review by iruzevic — APPROVED** [review]

**Review by mbmjertan — APPROVED** [review]

**Review by iobrado — APPROVED** [review]

**Pull request merged**